source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -332,6 +338,44 @@ def filter_func(self, journey):
         return True
 
 
+class FilterDirectPathMode(SingleJourneyFilter):
+
+    message = 'direct_path_mode_parameter'
+
+    def __init__(self, dp, dp_mode, orig_modes, dest_modes):
+        self.dp = dp
+        self.dp_mode = [] if dp_mode is None else dp_mode
+        self.orig_modes = [] if orig_modes is None else orig_modes
+        self.dest_modes = [] if dest_modes is None else dest_modes
+
+    def filter_func(self, journey):
+        """
+        eliminates journeys that are not matching direct path mode parameter
+        """
+
+        is_dp = 'non_pt' in journey.tags
+        is_in_direct_path_mode_list = any(mode in self.dp_mode for mode in journey.tags)
+        is_in_fallback_mode_list = any(mode in self.orig_modes for mode in journey.tags) or any(
+            mode in self.dest_modes for mode in journey.tags
+        )
+
+        if self.dp == 'none' and is_dp:
```

Comment:
Shouldn't it be managed by another filter, called in any case and mutualized?
(same below)

And then it's the integrator's problem if he asks for direct_path_modes[]=taxi&direct_path=none (resulting in no direct_path).